sunpy/map/mapbase.py

```diff
@@ -2187,11 +2189,33 @@ def contour(self, level, **kwargs):
         contours: list of (n,2) `~astropy.coordinates.SkyCoord`
             Coordinates of each contour.
 
+        Examples
+        --------
+        >>> import astropy.units as u
+        >>> from astropy.coordinates import SkyCoord
+        >>> import sunpy.map
+        >>> import sunpy.data.sample  # doctest: +REMOTE_DATA
+        >>> aia = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)  # doctest: +REMOTE_DATA
+        >>> contours = aia.contour(50000 * u.ct)  # doctest: +REMOTE_DATA
+        >>> print(contours[0])  # doctest: +REMOTE_DATA
+            <SkyCoord (Helioprojective: obstime=2011-06-07T06:33:02.770, rsun=696000000.0 m, observer=<HeliographicStonyhurst Coordinate (obstime=2011-06-07T06:33:02.770): (lon, lat, radius) in (deg, deg, m)
+        (-0.00406308, 0.04787238, 1.51846026e+11)>): (Tx, Ty) in arcsec
+        [(719.59798458, -352.60839064), (717.19243987, -353.75348121),
+        ...
+
         See also
         --------
         `skimage.measure.find_contours`
         """
         from skimage import measure
+
+        if self.unit is not None:
+            try:
+                level = level.to_value(self.unit)
+            # Catch cases where level is a float or can't be converted
```

Comment:
What's the issue if `level` is a float?  Should that part of the comment be removed?

Reply:
If `level` is a float but `self.unit != None` then this errors, since we want the user to give `level` as a `Quantity`. I'll improve the comment here to make it clearer.

Reply:
It does?  I just tried it (`aia.contour(50000.1*u.ct)`) and no error...  Am I misunderstanding something?

Reply:
I guess it depends on the type of the data array?

Reply:
`50000.1 * u.ct` should be a `Quantity`, not a `float` - I'm interpreting `float` in the strict sense of `isinstance(num, float) == True`.

Reply:
Oh!  So you really mean a non-`Quantity`, whether it's a `float`, `int`, or whatever.

Reply:
Yep, I have updated the comment to "not a Quantity" to be a bit clearer